v2/account_claims.go

```diff
@@ -250,14 +251,19 @@ func (a *AccountClaims) ClearRevocation(pubKey string) {
 	a.Revocations.ClearRevocation(pubKey)
 }
 
-// IsRevokedAt checks if the public key is in the revoked list with a timestamp later than
-// the one passed in. Generally this method is called with time.Now() but other time's can
-// be used for testing.
+// IsRevokedAt checks if the public key is in the revoked list with a timestamp later than the one passed in.
+// Generally this method is called with the subject and issue time of the jwt to be tested.
+// DO NOT pass time.Now(), it will not produce a stable/expected response.
+// Deprecated: use IsClaimRevokedAt
 func (a *AccountClaims) IsRevokedAt(pubKey string, timestamp time.Time) bool {
 	return a.Revocations.IsRevoked(pubKey, timestamp)
 }
 
-// IsRevoked checks if the public key is in the revoked list with time.Now()
-func (a *AccountClaims) IsRevoked(pubKey string) bool {
-	return a.Revocations.IsRevoked(pubKey, time.Now())
+// IsClaimRevoked checks if the account revoked the claim passed in.
+// Invalid claims (nil, no Subject or IssuedAt) will return true.
+func (a *AccountClaims) IsClaimRevoked(claim *UserClaims) bool {
+	if claim == nil || claim.IssuedAt == 0 || claim.Subject == "" {
```

Comment:
This logic seems wonky, if the claim has no subject, it should be illegal, so the decoding functionality should have failed. This reduces all sorts of complexities downstream.

If the claim is nil, not sure how we got here, also indicates a problem elsewhere

Reply:
You could have constructed the UserClaims in memory without calling decode. 
I return the safe value (true) when smth is off

Reply:
That has no implications outside of bad code.

Reply:
And we don't crash there either, just as go recommends.

Reply:
Yes - again, I think the issue is not that the code does it or not, but rather that now my question is why doesn't other code do the same checks. If the issue is catching internal tests, I don't really care for the extra "cannot happen" scenarios (server will decode all the JWTs and they are either valid or not). Two valid JWTs should be comparable without extra paranoia.

Reply:
DidSign checks if op is nil
Encode checks if an accounts subject has the right type. it also checks (further down) if the keyPair is nil.
activationClaims HashID performs similar checks.
NewOperatorClaims checks arguments

Reply:
I left the checks in due to all those other checks existing

Reply:
After discussion we agreed that it is ok to remove from v2 since it has not been released yet.

Reply:
(I meant IsRevoked() and IsRevokedAt(), not the new IsClaimRevoked() ;-) )